jina/peapods/runtimes/request_handlers/data_request_handler.py

```diff
@@ -132,18 +132,27 @@ def handle(
             self.logger.debug(
                 f'skip executor: mismatch request, exec_endpoint: {msg.envelope.header.exec_endpoint}, requests: {self._executor.requests}'
             )
+            if partial_requests:
+                DataRequestHandler.replace_docs(
```

Comment:
I think here u should do nothing right?

Reply:
Clarified offline. For the record:

The reason I added this: When you do a `Flow.add(name='1').add(name='2', needs['gateway', '1'])` by default it uses the `BaseExecutor` in the HeadPea of `2`. But the `BaseExecutor` has no endpoint defined. Anyhow, it should merge the the docs from both pathways to one big `DocumentArray`.